chart/kubeapps/templates/kubeapps-frontend-config.yaml

```diff
@@ -57,36 +57,42 @@ data:
       # Ensure each cluster can be reached (should only be
       # used with an auth-proxy where k8s credentials never leave
       # the cluster). See clusters option.
-      {{- range .Values.clusters }}
+    {{- range .Values.clusters }}
       location ~* /api/clusters/{{ .name }} {
-        rewrite /api/clusters/{{ .name }}/(.*) /$1 break;
-        rewrite /api/clusters/{{ .name }} / break;
-
-      {{/* Helm returns a nil pointer error when accessing foo.bar if foo doesn't
+    {{/* We need to split the API service URL(s) into the base url and the path segment so
+           those configurations using a path can be appropriately rewritten below while
+           ensuring the proxy_pass statement is given the base URL only. */}}
+    {{- $parsed := urlParse (default "" .apiServiceURL) }}
```

Comment:
@absoludity only thing I'm unsure about now is why the default value here is `''` instead of `'https://kubernetes.default'` like it [is/was before](https://github.com/kubeapps/kubeapps/pull/2337/files?w=1#diff-122c6e16fc44dcfed4ed0ece01fc53e3d16f46fa3fcd3ba138e61dc3891206a5L82)...

Shouldn't we do:

```
{{- $parsed := urlParse (default "https://kubernetes.default" .apiServiceURL) }}
```
?